### PR TITLE
Add a section to the coding guide on naming best practices

### PR DESCRIPTION
**Pull request by namurphy**
This PR is another spin-off from #1305 that focuses on best practices on naming variables and functions.  

This depends on a few things in #1649, and will need to be merged afterwards.

### CODE REVIEW

File: docs/contributing/code_guide.rst [hunk 1]
```diff
@@ -20,19 +20,144 @@ style changes. Please feel free to propose revisions to this guide by
...
+  Prefer clarity over brevity, except for code that is used frequently
+  and interactively (e.g., :command:`cd` or :command:`ls`).
+
+  .. tip::
+
+     Measure the length of a variable not by the number of characters,
+     but rather by the time needed to understand its meaning.
+
+     By this measure, ``cggglm`` is significantly longer than
+     ``solve_gauss_markov_linear_model``.
```
Comment (StanczakDominik) [inline docs/contributing/code_guide.rst:69]: 🤣 oh, the flashbacks!

File: docs/contributing/code_guide.rst [hunk 2]
```diff
@@ -20,19 +20,144 @@ style changes. Please feel free to propose revisions to this guide by
...
+  a commonly used symbol) and in Jupyter_ notebooks.
+
+* If a plasma parameter has multiple names, then use the name that
+  provides the most physical insight. For example, ``gyrofrequency``
+  indicates gyration but ``Larmor_frequency`` does not.
+
+* It is *usually* preferable to name a variable after its name rather
+  than its symbol.  An object named ``Debye_length`` is more broadly
+  understandable and searchable than ``lambda_D``. However, there are
+  some exceptions to this guideline.
```
Comment (StanczakDominik) [inline docs/contributing/code_guide.rst:93]: As with all guidelines... :)

File: docs/contributing/code_guide.rst [hunk 3]
```diff
@@ -20,19 +20,144 @@ style changes. Please feel free to propose revisions to this guide by
...
-* PlasmaPy follows the `PEP8 Style Guide for Python Code
-  <https://peps.python.org/pep-0008>`__.  This style choice
-  helps ensure that the code will be consistent and readable.
-
-  * Line lengths should be chosen to maximize the readability and
-    elegance of the code.  The maximum line length for Python code in
-    PlasmaPy is 88 characters.
+* PlasmaPy generally follows the :pep:`8` style guide for Python code.
+  This style choice helps ensure that the code will be consistent and
+  readable.
```
Comment (StanczakDominik) [inline docs/contributing/code_guide.rst:160]: Well, actually...
```suggestion
* PlasmaPy generally follows the :pep:`8` style guide for Python code and the black_ code style.
  This helps ensure that the code will be consistent and
  readable.
```

**Review by StanczakDominik — APPROVED** [review]
Good stuff!

**Comment by namurphy** [comment]
Thank you for the prompt and helpful review!

**Pull request merged**